primitives/state-machine/src/trie_backend_essence.rs

```diff
@@ -79,9 +104,19 @@ where
 
 	/// Set trie root. This is useful for testing.
 	pub fn set_root(&mut self, root: H::Out) {
+		// If root did change so can have cached content.
+		self.reset_cache();
```

Comment:
Don't we also need to reset the cache after having run `storage_root`?

Reply:
Should be fine, the cache is above the trie backend so bellow all overlay and ext stuff, it really only change when trie change (after client does process block when state machine instance is already dropped).
Here it is related to the way it was use in some test function (changing root means basically changing the trie backend).